src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/IssueFurtherEvidenceHandler.java

```diff
@@ -35,44 +47,72 @@ public IssueFurtherEvidenceHandler(FurtherEvidenceService furtherEvidenceService
+        List<DocumentType> documentTypes = Arrays.asList(APPELLANT_EVIDENCE, REPRESENTATIVE_EVIDENCE, DWP_EVIDENCE);
+        List<FurtherEvidenceLetterType> allowedLetterTypes = Arrays.asList(APPELLANT_LETTER, REPRESENTATIVE_LETTER);
+        documentTypes.forEach(documentType -> doIssuePerDocumentType(caseData, allowedLetterTypes, documentType));
+    }
 
-        setEvidenceIssuedFlagToYes(callback.getCaseDetails().getCaseData().getSscsDocument());
-        updateCase(callback.getCaseDetails().getCaseData());
+    private void doIssuePerDocumentType(SscsCaseData caseData, List<FurtherEvidenceLetterType> allowedLetterTypes,
+                                        DocumentType documentType) {
+        try {
+            furtherEvidenceService.issue(caseData.getSscsDocument(), caseData, documentType, allowedLetterTypes);
+        } catch (Exception e) {
+            handleIssueFurtherEvidenceException(caseData, documentType);
+            String errorMsg = "Failed sending further evidence for case(%s)...";
+            throw new IssueFurtherEvidenceException(String.format(errorMsg, caseData.getCcdCaseId()), e);
+        }
     }
 
-    private void setEvidenceIssuedFlagToYes(List<SscsDocument> sscsDocuments) {
+    private void postIssueFurtherEvidenceTasks(SscsCaseData caseData) {
+        try {
+            setEvidenceIssuedFlagToYes(caseData.getSscsDocument());
+            ccdService.updateCase(caseData, Long.valueOf(caseData.getCcdCaseId()),
+                EventType.UPDATE_CASE_ONLY.getCcdType(),
+                "Update case data",
+                "Update issued evidence document flags after issuing further evidence",
+                idamService.getIdamTokens());
+        } catch (Exception e) {
+            String errorMsg = "Failed to update document evidence issued flags after issuing further evidence "
+                + "for case(%s)";
+            throw new PostIssueFurtherEvidenceTasksException(String.format(errorMsg, caseData.getCcdCaseId()), e);
+        }
+    }
 
+    private void handleIssueFurtherEvidenceException(SscsCaseData caseData, DocumentType documentType) {
+        caseData.setHmctsDwpState("failedSendingFurtherEvidence");
+        ccdService.updateCase(caseData, Long.valueOf(caseData.getCcdCaseId()),
+            EventType.SEND_FURTHER_EVIDENCE_ERROR.getCcdType(), "Failed to issue further evidence",
+            "Review document tab to see document(s) that haven't been issued, then use the"
+                + " \"Reissue further evidence\" within next step and select affected document(s) to re-send",
+            idamService.getIdamTokens());
+    }
+
+    private void setEvidenceIssuedFlagToYes(List<SscsDocument> sscsDocuments) {
         if (sscsDocuments != null) {
             for (SscsDocument doc : sscsDocuments) {
-                if (doc.getValue().getEvidenceIssued() != null && doc.getValue().getEvidenceIssued().equals("No")) {
+                if (doc.getValue().getEvidenceIssued() != null
```

Comment:
Minor thing but do you need this NullPointer check now you have changed line 109?